test/blackbox/common/DDSBlackboxTestsPersistence.cpp

```diff
@@ -473,16 +472,110 @@ TEST_P(PersistenceLargeData, PubSubAsReliablePubPersistentWithStaticDiscovery)
+
+    writer
+            .history_kind(eprosima::fastdds::dds::KEEP_ALL_HISTORY_QOS)
+            .resource_limits_max_samples(100)
+            .reliability(eprosima::fastdds::dds::RELIABLE_RELIABILITY_QOS)
+    // A PERSISTENT writer with a persistence guid must behave as TRANSIENT
+            .make_persistent(db_file_name(), "77.72.69.74.65.72.5f.70.65.72.73.5f|67.75.69.64")
+            .init();
+
+    ASSERT_TRUE(writer.isInitialized());
+
+    auto data = default_helloworld_data_generator();
+    auto received_data = data;
+
+    // Send data
+    writer.send(data);
+    // All data should be sent
+    ASSERT_TRUE(data.empty());
+    // Destroy the DataWriter
+    writer.destroy();
+
+    reader
+            .history_kind(eprosima::fastdds::dds::KEEP_LAST_HISTORY_QOS)
+            .history_depth(10)
+            .reliability(eprosima::fastdds::dds::RELIABLE_RELIABILITY_QOS)
+    // A TRANSIENT reader with no persistence guid should behave as TRANSIENT_LOCAL
+            .durability_kind(eprosima::fastdds::dds::TRANSIENT_DURABILITY_QOS)
+            .init();
+
+    ASSERT_TRUE(reader.isInitialized());
+
+    // Load the transient DataWriter with the changes saved in the database
+    writer.init();
+
+    ASSERT_TRUE(writer.isInitialized());
+
+    // Wait for discovery.
+    writer.wait_discovery();
+    reader.wait_discovery();
+
+    reader.startReception(received_data);
+
+    // Wait expecting receiving all data.
+    reader.block_for_all();
+}
+
+TEST_P(DDSPersistenceTests, PubSubAsReliablePubTransientWithNoPersistenceGUIDBehavesAsTransientLocal)
+{
+    PubSubWriter<HelloWorldPubSubType> writer(TEST_TOPIC_NAME);
+    PubSubReader<HelloWorldPubSubType> reader(TEST_TOPIC_NAME);
+
+    writer
+            .history_kind(eprosima::fastdds::dds::KEEP_ALL_HISTORY_QOS)
+            .resource_limits_max_samples(100)
+            .reliability(eprosima::fastdds::dds::RELIABLE_RELIABILITY_QOS)
+    // A TRANSIENT writer with a persistence guid must behave as TRANSIENT_LOCAL
+            .durability_kind(eprosima::fastdds::dds::TRANSIENT_DURABILITY_QOS)
+            .init();
+
+    ASSERT_TRUE(writer.isInitialized());
+
+    auto data = default_helloworld_data_generator();
+    auto received_data = data;
+
+    // Send data
+    writer.send(data);
+
+    // All data should be sent
+    ASSERT_TRUE(data.empty());
+
+    reader
+            .history_kind(eprosima::fastdds::dds::KEEP_LAST_HISTORY_QOS)
+            .history_depth(10)
+            .reliability(eprosima::fastdds::dds::RELIABLE_RELIABILITY_QOS)
+    // A TRANSIENT reader with no persistence guid should behave as TRANSIENT_LOCAL
+            .durability_kind(eprosima::fastdds::dds::TRANSIENT_DURABILITY_QOS)
+            .init();
+
+    ASSERT_TRUE(reader.isInitialized());
+
+    // Wait for discovery.
+    writer.wait_discovery();
+    reader.wait_discovery();
+
+    reader.startReception(received_data);
+
+    // Wait expecting receiving all data.
+    reader.block_for_all();
```

Comment:
After this part, it would be nice to check that re-creating the writer and the reader results in the reader receiving nothing.
This way we would check that the writer is not transient.